bpf/lib/overloadable_skb.h

```diff
@@ -122,24 +122,20 @@ ctx_set_cluster_id_mark(struct __sk_buff *ctx, __u32 cluster_id)
 	__u32 cluster_id_lower = (cluster_id & 0xFF);
 	__u32 cluster_id_upper = ((cluster_id & 0xFFFFFF00) << (8 + IDENTITY_LEN));
 
-	ctx->mark &= MARK_MAGIC_KEY_MASK;
-	ctx->mark |=  cluster_id_lower | cluster_id_upper | MARK_MAGIC_CLUSTER_ID;
+	ctx->mark = cluster_id_lower | cluster_id_upper | MARK_MAGIC_CLUSTER_ID;
 }
 
 static __always_inline __maybe_unused __u32
-ctx_get_cluster_id_mark(struct __sk_buff *ctx)
+ctx_get_cluster_id_mark(const struct __sk_buff *ctx)
 {
 	__u32 ret = 0;
 	__u32 cluster_id_lower = ctx->mark & CLUSTER_ID_LOWER_MASK;
 	__u32 cluster_id_upper = (ctx->mark & get_cluster_id_upper_mask()) >> (8 + IDENTITY_LEN);
 
-	if ((ctx->mark & MARK_MAGIC_CLUSTER_ID) != MARK_MAGIC_CLUSTER_ID)
+	if ((ctx->mark & MARK_MAGIC_HOST_MASK) != MARK_MAGIC_CLUSTER_ID)
 		return ret;
```

Comment:
```suggestion
	if ((ctx->mark & MARK_MAGIC_HOST_MASK) != MARK_MAGIC_CLUSTER_ID)
		return 0;
```

A little less brain-twisting this way :)

Reply:
It looks like the change requires to mark the `ret` variable as `__maybe_unused`, otherwise the compiler complains: https://github.com/cilium/cilium/actions/runs/20129215113/job/57766211539?pr=43258. Let me know if you prefer the new version, or I should revert it to the previous one.

Reply:
well because it **is** unused now, right? Your second patch removed the other usage of it :smile:

Reply:
🤦 (Let me fix that)